packages/analytics/amplify_analytics_pinpoint/example/integration_test/events_test.dart

```diff
@@ -0,0 +1,132 @@
+    // There is some lossiness behavior to doubles
+    const lossyDoubleProperty = MapEntry('doubleProperty', 3.141592653589793);
+
+    const intProperty = MapEntry('intProperty', 42);
+    const stringProperty = MapEntry('stringProperty', 'Hello, world');
+
+    late final Stream<Map<String, Object?>> eventsStream;
+
+    setUpAll(() async {
+      await Amplify.addPlugins([
+        AmplifyAuthCognito(),
+        AmplifyAnalyticsPinpoint(),
+        AmplifyAPI(),
+      ]);
+      await Amplify.configure(amplifyconfig);
+
+      eventsStream = Amplify.API
+          .subscribe(
+        GraphQLRequest<String>(document: '''
+subscription {
+  onCreateRecord {
+    id
+    payload
+  }
+}
+'''),
+      )
+          .map((event) {
+        if (event.errors.isNotEmpty) {
+          return {'errors': event.errors};
+        }
+        final data = event.data;
+        if (data == null) {
+          return const <String, Object?>{};
+        }
+        final json = jsonDecode(data) as Map<String, Object?>;
+        return jsonDecode((json['onCreateRecord'] as Map)['payload'] as String)
+            as Map<String, Object?>;
+      }).asBroadcastStream();
+
+      eventsStream.listen((event) {
+        print('Got event: ${jsonEncode(event)}');
+      });
+    });
+
+    test(
+      'can record custom event',
+      () async {
+        final customEvent = AnalyticsEvent('MyCustomEvent');
+
+        customEvent.properties
+          ..addBoolProperty(boolProperty.key, boolProperty.value)
+          ..addDoubleProperty(doubleProperty.key, doubleProperty.value)
+          ..addIntProperty(intProperty.key, intProperty.value)
+          ..addStringProperty(stringProperty.key, stringProperty.value);
+
+        expect(
+          customEvent.properties.getAllProperties(),
+          Map.fromEntries([
+            boolProperty,
+            doubleProperty,
+            intProperty,
+            stringProperty,
+          ]),
+        );
+        expect(
+          customEvent.properties.getAllPropertiesTypes(),
+          {
+            boolProperty.key: 'BOOL',
+            doubleProperty.key: 'DOUBLE',
+            intProperty.key: 'INT',
+            stringProperty.key: 'STRING',
+          },
+        );
+
+        expect(
+          eventsStream,
+          emitsThrough(allOf([
+            containsPair('event_type', 'MyCustomEvent'),
+            containsPair(
+              'attributes',
+              // Boolean properties are recorded as strings
+              Map.fromEntries([stringifiedBoolProperty, stringProperty]),
+            ),
+            containsPair(
+              'metrics',
+              // There is some lossiness behavior to doubles
+              Map.fromEntries([lossyDoubleProperty, intProperty]),
+            ),
+          ])),
+        );
+
+        await Amplify.Analytics.recordEvent(event: customEvent);
+        await Amplify.Analytics.flushEvents();
+      },
+      timeout: const Timeout(Duration(minutes: 1)),
```

Comment:
nit/suggestion: I got some timeouts when trying this and raising to 2 seemed to help 😢 .